### PR TITLE
18619 20719 eft statement updates

### PR DESCRIPTION
**Pull request by ochiu**
Issue #:
https://github.com/bcgov/entity/issues/18619
https://github.com/bcgov/entity/issues/20719

*Description of changes:*
- update account statements route to use eft_statement template and implementing input calculation logic
- added in is_interim_statement flag to statement model for statement template output logic
- some refactor to make code re-usable and easier to maintain
- updates to eft statement template with ticket updates and include interim statement variation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Comment by ochiu** [comment]
Rebase of https://github.com/bcgov/sbc-pay/pull/1549 onto main due to conflicts.

Additional changes was some UX fixes and splitting the eft template into separate files

File: pay-admin/pyproject.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jinja2 = "^3.1.3"
 pay-api = {git = "https://github.com/bcgov/sbc-pay.git", subdirectory = "pay-api"}
 sbc-common-components = {git = "https://github.com/bcgov/sbc-common-components.git", subdirectory = "python"}
 simple-cloudevent = {git = "https://github.com/daxiom/simple-cloudevent.py.git"}
-flask-jwt-oidc = {git = "https://github.com/thorwolpert/flask-jwt-oidc.git"}
```
Comment (seeker25) [inline pay-admin/pyproject.toml:28]: I'll clean this up today, don't need any of these just `pay-api` it already includes all of these

File: pay-api/src/pay_api/services/report_service.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    populate_page_number: bool
+
+
+class ReportService:
+    """Service to manage report generation."""
+
+    @staticmethod
+    def get_request_payload(request: ReportRequest) -> dict:
+        """Return report request payload dictionary."""
+        return {
```
Comment (seeker25) [inline pay-api/src/pay_api/services/report_service.py:40]: just return `humps.camelize(asdict(request))`

similar to 
image

File: pay-api/src/pay_api/services/statement.py [hunk 3]
```diff
@@ -138,12 +144,67 @@ def find_by_account_id(auth_account_id: str, page: int, limit: int):
         return data
 
     @staticmethod
-    def get_statement_report(statement_id: str, content_type: str, template_name='statement_report', **kwargs):
+    def get_statement_template(statement: StatementModel, statement_invoices: List[InvoiceModel]) -> str:
```
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:147]: suggestion: `ordered_statement_invoices` or `ordered_invoices`

File: pay-api/src/pay_api/services/report_service.py (hunk 2 above)
Comment (ochiu) [reply pay-api/src/pay_api/services/report_service.py:40]: There is an issue with his as the outer report payload expects camel case, but a lot of the actual template_vars are snake case and used as such in existing templates due to how the values are retrieved.

Comment (seeker25) [reply pay-api/src/pay_api/services/report_service.py:40]: I still don't see the issue, you're expecting camelCase in this scenario

Comment (ochiu) [reply pay-api/src/pay_api/services/report_service.py:40]: I believe it converts the whole payload including statement attributes to camel as well, but existing templates use statement.from_date

```
"templateVars": {
  "statement": {
              "created_on": "2024-05-27",
              "frequency": "MONTHLY",
              "from_date": "2024-04-01",
              "to_date": "2024-04-30",
              ...
  }
```

**Review by seeker25 — APPROVED** [review]

File: pay-api/src/pay_api/services/report_service.py (hunk 2 above)
Comment (seeker25) [reply pay-api/src/pay_api/services/report_service.py:40]: I see what you mean now and that's a huge pain to fix everywhere along the way

File: pay-api/src/pay_api/services/statement.py (hunk 3 above)
Comment (ochiu) [reply pay-api/src/pay_api/services/statement.py]: Updated to ordered_invoices

**Pull request merged**